### PR TITLE
The third parameter of Driver::executeQuery is now an options array

### PR DESCRIPTION
**Pull request by timwhitlock**

### CODE REVIEW

File: mongodb/mongodb.php [hunk 1]
```diff
@@ -352,14 +352,14 @@ final class Server
...
-             * @param ReadPreference $readPreference Optionally, a MongoDB\Driver\ReadPreference to select the server for this operation. If none is given, the read preference from the MongoDB Connection URI will be used.
+             * @param array|ReadPreference $options The third parameter is now an options array. For backwards compatibility, this parameter will still accept a MongoDB\Driver\ReadPreference object.
              * @throws InvalidArgumentException on argument parsing errors.
              * @throws ConnectionException if connection to the server fails (for reasons other than authentication).
              * @throws AuthenticationException if authentication is needed and fails.
              * @throws RuntimeException on other errors (e.g. invalid command, issuing a write command to a secondary).
              * @return Cursor
              */
-            final public function executeQuery($namespace, Query $query, ReadPreference $readPreference = null)
+            final public function executeQuery($namespace, Query $query, array $option = [])
```
Comment (andrey-sokolov) [inline mongodb/mongodb.php:362]: ok let's omit the array type-hint here to make it consistent with PhpDoc

File: mongodb/mongodb.php [hunk 2]
```diff
@@ -352,14 +352,14 @@ final class Server
              * @link https://php.net/manual/en/mongodb-driver-server.executequery.php
              * @param string $namespace A fully qualified namespace (e.g. "databaseName.collectionName").
              * @param Query $query The MongoDB\Driver\Query to execute.
-             * @param ReadPreference $readPreference Optionally, a MongoDB\Driver\ReadPreference to select the server for this operation. If none is given, the read preference from the MongoDB Connection URI will be used.
+             * @param array|ReadPreference $options The third parameter is now an options array. For backwards compatibility, this parameter will still accept a MongoDB\Driver\ReadPreference object.
```
Comment (andrey-sokolov) [inline mongodb/mongodb.php:355]: we need to annotate the parameter with up-to-date info from php.net as well

File: mongodb/mongodb.php (hunk 1 above)
Comment (andrey-sokolov) [inline mongodb/mongodb.php:362]:
```suggestion
            final public function executeQuery($namespace, Query $query, $option = [])
```

File: mongodb/mongodb.php (hunk 2 above)
Comment (andrey-sokolov) [inline mongodb/mongodb.php:355]:
```suggestion
             * @param array|ReadPreference $options 
             * <table>
             * <caption><strong>options</strong></caption>
             *
             * <thead>
             * <tr>
             * <th>Option</th>
             * <th>Type</th>
             * <th>Description</th>
             * </tr>
             *
             * </thead>
             *
             * <tbody>
             *
             * <tr>
             * <td>readPreference</td>
             * <td><a href="https://php.net/manual/en/php.neclass.mongodb-driver-readpreference.php">MongoDB\Driver\ReadPreference</a></td>
             * <td>
             * <p>
             * A read preference to use for selecting a server for the operation.
             * </p>
             * </td>
             * </tr>
             * <tr>
             * <td>session</td>
             * <td><a href="https://php.net/manual/en/class.mongodb-driver-session.php">MongoDB\Driver\Session</a></td>
             * <td>
             * <p>
             * A session to associate with the operation.
             * </p>
             * </td>
             * </tr>
             * </tbody>
             * </table>
             * The third parameter is now an options array. For backwards compatibility, this parameter will still accept a MongoDB\Driver\ReadPreference object.
```

**Comment by timwhitlock** [comment]
I didn't realise your comments were aimed at me. Yes looks fine.

File: mongodb/mongodb.php [hunk 3]
```diff
@@ -352,14 +352,49 @@ final class Server
              * @link https://php.net/manual/en/mongodb-driver-server.executequery.php
              * @param string $namespace A fully qualified namespace (e.g. "databaseName.collectionName").
              * @param Query $query The MongoDB\Driver\Query to execute.
-             * @param ReadPreference $readPreference Optionally, a MongoDB\Driver\ReadPreference to select the server for this operation. If none is given, the read preference from the MongoDB Connection URI will be used.
+             * @param array|ReadPreference $options
```
Comment (andrey-sokolov) [inline mongodb/mongodb.php:355]:
```suggestion
             * @param array|ReadPreference $options
```

**Comment by andrey-sokolov** [comment]
@timwhitlock Thank you for the contribution

**Pull request merged**